eng/pipelines/libraries/run-test-job.yml

```diff
@@ -7,6 +7,8 @@ parameters:
   isOfficialBuild: false
   liveRuntimeBuildConfig: ''
   runtimeFlavor: 'coreclr'
+  runtimeDisplayName: 'coreclr'
```

Comment:
why is this necessary? We already have runtimeFlavorName: https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/xplat-setup.yml#L53

Reply:
The way we're using runtimeFlavor seemed too broad.  We need a separate mono interpreter test lane, but 'interpreter' is not a runtime flavor.  If you add another and just keep it mono, then the jobs will fail b/c the names will be the same.

Reply:
`runtimeDisplayName` feels like the wrong name to me, I'd rather call it something like `testConfig` which could be mono-interpreter, mono-fullaot, mono-...